master/internal/sproto/task.go

```diff
@@ -121,25 +123,45 @@ type (
 	ResourcesAllocated struct {
 		ID           model.AllocationID
 		ResourcePool string
-		Reservations []Reservation
+		Resources    []Resources
 	}
 	// ReleaseResources notifies the task actor to release resources.
 	ReleaseResources struct {
 		ResourcePool string
 	}
-	// ReservationRuntimeInfo is all the inforamation provided at runtime to make a task spec.
-	ReservationRuntimeInfo struct {
+	// ResourcesRuntimeInfo is all the inforamation provided at runtime to make a task spec.
+	ResourcesRuntimeInfo struct {
 		Token        string
 		AgentRank    int
 		IsMultiAgent bool
 	}
 )
 
-// Reservation is an interface that provides function for task actors
+const (
+	// ResourcesTypeK8sPod indicates the resources are a handle for a k8s pod.
+	ResourcesTypeK8sPod ResourcesType = "k8s-pod"
+	// ResourcesTypeDockerContainer indicates the resources are a handle for a docker container.
+	ResourcesTypeDockerContainer ResourcesType = "docker-container"
+)
+
+// ResourcesSummary provides a summary of the resources comprising what we know at the time the
+// allocation is granted, but for k8s it is granted before being scheduled so it isn't really much
+// and `agent_devices` are missing for k8s.
+type ResourcesSummary struct {
+	ResourcesID   ResourcesID                `json:"resources_id"`
+	ResourcesType ResourcesType              `json:"resources_type"`
+	AllocationID  model.AllocationID         `json:"allocation_id"`
+	AgentDevices  map[string][]device.Device `json:"agent_devices"`
```

Comment:
perhaps `type AgentID string` and
```suggestion
	AgentDevices  map[AgentID][]device.Device `json:"agent_devices"`
```